### ISSUE TITLE
[Bug]: Colorbar does not work for negative values with contour/contourf

### ISSUE THREAD

**Issue by sodlr** [matplotlib/matplotlib#21882]
### Bug summary

The colorbar does not work with contour/contourf, if the data contains negative values AND zeros.
The error message is the following:

matplotlib/transforms.py:2663: RuntimeWarning: divide by zero encountered in double_scalars
  y_scale = 1.0 / inh
matplotlib/transforms.py:2665: RuntimeWarning: invalid value encountered in double_scalars
  [0.0    , y_scale, (-inb*y_scale)]

It does work for only negative (without zero), only positive (without zero), and mixed positive and negative values.
The colorbar also works with imshow regardless of the data.

### Code for reproduction

```python

import matplotlib.pyplot as plt
import numpy as np

# mixed positive and negative values work
data = np.arange(-5, 4).reshape((3,3))

fig, ax = plt.subplots()
im = ax.contourf(data)
cb = plt.colorbar(im)

# negative and zero does not work!
data = np.arange(-5, 4).reshape((3,3))
vmax = np.max(data)
data -= vmax

fig, ax = plt.subplots()
im = ax.contourf(data)
cb = plt.colorbar(im)
```

### Actual outcome

First example that works!

![Figure 2021-12-08 161957](https://user-images.githubusercontent.com/95762854/145234105-7bf26e26-562e-46d2-887d-96a95709e45d.png)

Second example that does not work!

![Figure 2021-12-08 162009](https://user-images.githubusercontent.com/95762854/145234138-d203525a-47dd-4e1c-bc81-49c21177d846.png)

### Expected outcome

The same plot as the first example with a colorbar from 0 to -9, which is produced with matplotlib 3.4.3.

### Matplotlib Version

3.5.0

**Comment by jklymak** [comment]
Yes, sorry this broke.  It was fixed in  #21676 and reported in #21671.  A fix will be out in 3.5.1 (hopefully soon)

**Comment by sodlr** [comment]
Thank you!
